### ISSUE TITLE
Disable default OUTPUT selects for mssql 

### ISSUE THREAD

**Issue by arnas** [typeorm/typeorm#2311]
**Issue type:**

[ ] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[x] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

Hello,
in the application I am developing I am required to use triggers ( more precise azure data sync that adds triggers on selected database tables). 
But as table has triggers queries fails if it has OUTPUT clause: "DML statement cannot have any enabled triggers if the statement contains an OUTPUT clause without INTO clause.". And it seems that this clause are added on default on every repository save, update, insert if entity has properties with CreateDateColumn or UpdateDateColumn decorators ( even select: false doesn't affect this).
I found I way to disable it in the driver configuration by changing function `        this.connection.driver.isReturningSqlSupported = () => false;`. But then typeorm does many one by one select after initial operation even if don't need any results. 
Maybe it would be good to have additional setting to disable output altogether or add to specif columns that I don't want this default behavior?

**Comment by pleerock** [comment]
output isn't reasonless there. If we add option to disable them then some functionality like CreateDateColumn and UpdateDateColumn won't work completely properly. Maybe in your very specific case there is another solution? Of if there is no other solutions maybe its better not to use CreateDateColumn and UpdateDateColumn decorators in your case?

**Comment by arnas** [comment]
Could you explain why output is required for CreateDateColumn ? As far I know it just adds Current_timestamp to a createdOn column ?

**Comment by pleerock** [comment]
because `save` method updates original entity you pass - updates all properties with DEFAULT set, so you have data in your model reflected from the database.

**Comment by arnas** [comment]
I see. So yeah global change could result in some weird behavior. But maybe there could be additional option for these column that are enabled be default?

**Comment by pleerock** [comment]
All additional options gonna look weird as well. Maybe its better to get know why mssql does not allow using output in your cases and how this situation can be handled?

**Comment by arnas** [comment]
Yeah I agree about that adding additional options can be an unnecessary overhead.
There is some workarounds about it https://stackoverflow.com/a/13198551. Maybe these could work with typeorm too. For example now if there is isReturningSqlSupported changed to false. Typeorm doesn't add OUTPUT, but fires many selects afterwards. So maybe instead of it it could insert to temp table and select afterwards if there is batch insert? Also maybe this behavior could be used in those databases that do not support `isReturningSqlSupported`.
What is your opinion about this?

**Comment by pleerock** [comment]
>  So maybe instead of it it could insert to temp table and select afterwards if there is batch insert?

definitely no.

The simplest solution I see is to add flag "disableOutputReturning" into mssql connection options and make a check of it inside isReturningSqlSupported of the driver.

**Comment by pleerock** [comment]
> the many select requests that are fired afterwards.

its not an issue. Selects are there because this is how `save` works - it sets all missing data into your model.

**Comment by kishore-indraganti** [comment]
is there any update on this, I am trying to implement typeorm package in my application. All the tables in mssql table have triggers, is there any way where I can prevent output options in generated insert query(both save and querybuilder method)

**Comment by pleerock** [comment]
lets ask @arnas if he solved this issue

**Comment by arnas** [comment]
Like i said in the first comment, i overwrited isReturningSqlSupported function.  
```
this.connection = await createConnection({ ...defaultOptions, ...customOptions });
this.connection.driver.isReturningSqlSupported = () => false;
```
I hope that helps

**Comment by pleerock** [comment]
hack =(

We could have a better solution via option just like I suggested. Isn't it better to create a PR?)

**Comment by arnas** [comment]
Seems good, I will try to do it this week

**Comment by pburrows** [comment]
Seems to me TypeORM should do this the same way Entity Framework does it (as mentioned here: https://stackoverflow.com/a/13198916/22399)

> Workaround #3 - Final
> In order to retrieve the after values, Entity Framework issues:
> 
> UPDATE [dbo].[BatchReports]
> SET [IsProcessed] = @0
> WHERE (([BatchReportGUID] = @1) AND ([RowVersion] = @2))
> 
> SELECT [RowVersion], [LastModifiedDate]
> FROM [dbo].[BatchReports]
> WHERE @@ROWCOUNT > 0 AND [BatchReportGUID] = @1
> Don't use OUTPUT.
> 
> Yes it suffers from a race condition, but that's the best SQL Server can do.
> 


Was a PR ever made for this? I don't see it linked. I'm willing to help work on it if needed.

**Comment by pburrows** [comment]
After spending an hour or so looking at this, it strikes me that TypeOrm won't be able to use the same technique that Entity Framework does because TypeOrm needs to have only 1 query executed for a given operation (insert, for example).

If this isn't the case, let me know. Otherwise, I am going to give up on this for now.

(also, it should be noted that the HACK above disables returning IDs for newly inserted entities, so it breaks a lot of basic functionality in TypeOrm).

**Comment by pleerock** [comment]
fixed via #2911 will be released in next 0.2.9 version.

**Comment by pburrows** [comment]
Also, if this issue is closed, can we add a separate issue to track doing this the right way? (as I mentioned in my comments above?)

This "fix" pretty much breaks TypeOrm. Doing an UPDATE and then a SELECT to get the resulting values is the best way to do this in SQL Server.
